airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/products_report.py

```diff
@@ -275,6 +284,27 @@ def _download_report(self, report_info: ReportInfo, url: str) -> List[dict]:
         """
         return super()._download_report(None, url)
 
+    @backoff.on_exception(
+        backoff.expo,
+        (
+            requests.exceptions.Timeout,
+            requests.exceptions.ConnectionError,
+            TooManyRequests,
+        ),
+        max_tries=10,
+    )
+    def _send_http_request(self, url: str, profile_id: int, json: dict = None):
+        headers = self._get_auth_headers(profile_id)
+        if json:
+            response = self._session.post(url, headers=headers, json=json)
+        else:
+            # using session without auth as API returns 400 bad request if Authorization header presents in request
+            # X-Amz-Algorithm and X-Amz-Signature query params already present in the url, that is enough to make proper request
+            response = self._no_auth_session.get(url, headers=headers)
```

Comment:
I find this a little surprising. Shouldn't we inject a different authenticator in the stream instead of using separate session field?

Reply:
for this stream authentication is required for report creation, post request above.